SmartDeviceLink/private/SDLPresentAlertOperation.m

```diff
@@ -109,7 +109,9 @@ - (void)start {
     dispatch_group_enter(uploadFilesTask);
 
     dispatch_group_enter(uploadFilesTask);
-    [self sdl_uploadImagesWithCompletionHandler:^{
+    __block NSSet<NSString *> *uploadedArtworks = nil;
+    [self sdl_uploadImagesWithCompletionHandler:^(NSSet<NSString *> *__nullable artworkNames) {
+        uploadedArtworks = artworkNames;
```

Comment:
You brought up an interesting question when you asked what happens if the image upload fails?
1. We can't use `fileNeedsUpload` if the developer is using their own `name` and set `overwrite` to `true` for the artwork. This is because checking `fileNeedsUpload` will return `true` if an image with the same name was already uploaded. 
2. We could use `hasUploadedFile` but would also need to check if the developer used a static icon. It also would not account for the situation where an artwork with the same name has been uploaded but the second artwork with same name fails to upload. When we check `hasUploadedFile`, it will return `true` (because there is a file with the `name` on the remote, it just hasn't been overwritten with the new artwork). This means that wrong artwork will be shown for the alert icon.

Reply:
I think that's fine. It's a "known issue" but one that is unlikely to occur and the complication that the fix adds to the code is not worth the low risk issue right now IMO.

Reply:
Ok. Requested changes were made. Not sure what you mean in the first comment about checking for `supportsSoftButtonImages`. Do you want me to not upload that button at all if the image upload fails or if the button type is "image" should I change the type to "text" and add some default text in there?

Reply:
No, we can keep doing what we do with the soft button manager in just trying to send them as-is with the image RPC. We may want to re-visit that in the future or provide the developer with a configuration option of what to do (e.g. skip it entirely if it's image-only, don't show any soft buttons if any fail, etc.)